src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -571,24 +574,38 @@ private async Task<IEnumerable<DiagnosticApiResponse>> ListDetectorsInternal(Run
             SearchResults searchResults = null;
             if (queryText != null && queryText.Length > 3)
             {
-                var resourceParams = _internalApiHelper.GetResourceParams(context.OperationContext.Resource as IResourceFilter);
-                var res = await _searchService.SearchDetectors(context.OperationContext.RequestId, queryText, resourceParams);
-                string resultContent = await res.Content.ReadAsStringAsync();
-                searchResults = JsonConvert.DeserializeObject<SearchResults>(resultContent);
-                searchResults.Results = searchResults.Results.Where(x => x.Score > 0.3).ToArray();
-                allDetectors.ForEach(p =>
+                if (_searchService.IsEnabled())
                 {
-                    var det = (searchResults != null) ? searchResults.Results.FirstOrDefault(x => x.Detector == p.EntryPointDefinitionAttribute.Id) : null;
-                    if (det != null)
+                    var resourceParams = _internalApiHelper.GetResourceParams(context.OperationContext.Resource as IResourceFilter);
+                    var res = await _searchService.SearchDetectors(context.OperationContext.RequestId, queryText, resourceParams);
+                    if (res != null && res.Content != null)
                     {
-                        p.EntryPointDefinitionAttribute.Score = det.Score;
+                        string resultContent = await res.Content.ReadAsStringAsync();
+                        searchResults = JsonConvert.DeserializeObject<SearchResults>(resultContent);
+                        searchResults.Results = searchResults.Results.Where(x => x.Score > 0.3).ToArray();
+                        allDetectors.ForEach(p =>
+                        {
+                            var det = (searchResults != null) ? searchResults.Results.FirstOrDefault(x => x.Detector == p.EntryPointDefinitionAttribute.Id) : null;
+                            if (det != null)
+                            {
+                                p.EntryPointDefinitionAttribute.Score = det.Score;
+                            }
+                            else
+                            {
+                                p.EntryPointDefinitionAttribute.Score = 0;
+                            }
+                        });
+                        allDetectors = allDetectors.Where(x => x.EntryPointDefinitionAttribute.Score > 0).ToList();
                     }
                     else
                     {
-                        p.EntryPointDefinitionAttribute.Score = 0;
+                        throw new Exception("Search API returned null");
                     }
-                });
-                allDetectors = allDetectors.Where(x => x.EntryPointDefinitionAttribute.Score > 0).ToList();
+                }
+                else
+                {
+                    throw new Exception("Search service is disabled.");
```

Comment:
This would still break the local environment if we disable search service.

I have a branch with changes to disable search service - https://github.com/Azure/Azure-AppServices-Diagnostics/blob/c91f0cc20707bfa82d6d53d1d9ef8aae8899f634/src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs